database/src/main/java/database/dao/mysql/MySQLConnectionPool.java

```diff
@@ -137,12 +157,14 @@ private MysqlDataSource createDataSource(Settings settings) {
 
     private void initializePool() {
         for (int i = 0; i < initialPoolSize; i++) {
-            try {
-                connectionPool.add(createNewConnection());
-                currentConnections++;
-            } catch (SQLException e) {
-                LOG.error(UNABLE_CREATE_CONNECTION + ": {}", e.getMessage());
-                throw new UnableCreateConnectionException(UNABLE_CREATE_CONNECTION + ": " + e.getMessage());
+            synchronized (this) {
```

Comment:
`initializePool` method is executed only inside constructor, no need to make it synchronized.